api/client/logs.go

```diff
@@ -37,8 +37,8 @@ func (cli *DockerCli) CmdLogs(args ...string) error {
 		return err
 	}
 
-	if logType := c.HostConfig.LogConfig.Type; logType != "json-file" {
-		return fmt.Errorf("\"logs\" command is supported only for \"json-file\" logging driver (got: %s)", logType)
+	if logType := c.HostConfig.LogConfig.Type; logType != "json-file" && logType != "journald" {
```

Comment:
Hmm, it looks like we still need to read the Tty field of the container config, in order to know whether or not we need to demux the data that's coming back.  And in cases where we would need to demux the data, but the logdriver implementation doesn't support reading on the server, I think the error that we currently send back isn't muxed, and that'll confuse the client.  I think that's how we got the "Unrecognized input header" error that @LK4D4 was seeing when the server was built without the libraries.